simp_le.py

```diff
@@ -1469,11 +1469,17 @@ def persist_new_data(args, existing_data):
                     'This client is unable to solve any other challenges.')
 
     for name, auth in six.iteritems(authorizations):
-        challb = supported_challb(auth)
-        response, validation = challb.response_and_validation(client.net.key)
-        save_validation(roots[name], challb, validation)
-
-        client.answer_challenge(challb, response)
+        for _ in range(5):
+            time.sleep(1)
+            auth, _ = client.poll(auth)
+            if auth.body.status == messages.STATUS_VALID:
```

Comment:
Maybe we should check for the specific error where retrying is expected, rather than retrying on all errors?

Reply:
There is actually no error checking involved, the code polls the auth body and wait five seconds for it to be anything else than `messages.STATUS_VALID`, proceed with challenge response and validation if this happens, break the loop if it does not.

I drafted this in a hurry a while ago when we switched to ACME v2, now that I re-think about it would make a lot more sense to wait for `messages.STATUS_PENDING` and error gracefully if it does not happen under the chosen duration.

Reply:
Maybe the following would be better:
- check first for `messages.STATUS_VALID`, break loop if true
- then check for `messages.STATUS_PENDING`, proceed with challenge response and validation if true
- if neither become true by the end of chosen duration, error gracefully

Reply:
SGTM.